### DISCUSSION TITLE
Stream response

### DISCUSSION THREAD

**Discussion by muellerj** [jeremyevans/roda#232] [General]
I would like to return a `ZipTricks::Streamer` as body of the response. The object responds to `#each` and should double as a valid Rack response. How can tell Roda to allow the body (not being a string, I currently get the following error)

```

Roda::RodaError
unsupported block result: [200, {"Content-Length"=>"70244265", "Content-Type"=>"binary/octet-stream", "Content-Disposition"=>"attachment; filename=tc-photos.zip"}, #<ZipTricks::OutputEnumerator:0x0000007fa1f268d8 @streamer_options={}, @bufsize=65536, @archiving_block=#<Proc:0x0000007fa1f26798>>
```

Documentation here: https://github.com/WeTransfer/zip_tricks

**Comment by jeremyevans** [comment]
You should be able to do: `r.halt(response.finish_with_body(obj))`.

**Comment by muellerj** [comment]
Indeed! Thank you for the prompt reply ❤️
